packages/main/src/backend/export/outputVendors/ynab/ynab.ts

```diff
@@ -119,12 +119,20 @@ function convertTransactionToYnabFormat(originalTransaction: EnrichedTransaction
     category_id: getYnabCategoryIdFromCategoryName(originalTransaction.category),
     memo: originalTransaction.memo,
     cleared: ynab.SaveTransaction.ClearedEnum.Cleared,
+    import_id: buildImportId(originalTransaction), // [date][amount][description]
     // "approved": true,
     // "flag_color": "red",
     // "import_id": buildImportId(originalTransaction.description, amount, date) // 'YNAB:[milliunit_amount]:[iso_date]:[occurrence]'
   };
 }
 
+function buildImportId(transaction: EnrichedTransaction): string {
+  return `${transaction.date.substring(0, 10)}${transaction.chargedAmount}${transaction.description}`.substring(
+    0,
+    MAX_YNAB_IMPORT_ID_LENGTH,
+  );
+}
+
```

Comment:
because YNAB _import_id_ is 36 characters max and should be unique
`calculateTransactionHash` is variable length and truncated will miss some information